bin/yb-ctl

```diff
@@ -1402,11 +1530,20 @@ class ClusterControl:
         self.creating_cluster = True
         server_counts = self.options.replication_factor
         self.set_master_addresses()
-        self.cluster_config = {
-            "enable_ysql": self.options.enable_ysql,
-            "num_drives": self.options.num_drives,
-            "replication_factor": self.options.replication_factor
-        }
+        self.cluster_config = {}
+        self.options.set_cluster_config(self.cluster_config)
+
+        for attr_name in (
+            "enable_ysql",
+            "num_drives",
+            "replication_factor",
+            "ysql_port",
+            "ycql_port",
+            "yedis_port",
+        ):
+            attr_value = getattr(self.options, attr_name)
```

Comment:
this will throw if the attr does not exist, do we want that?

Reply:
These fields always exist.